soccer/planning/RRTPlanner.hpp

```diff
@@ -38,63 +38,62 @@ namespace Planning
 			{
 				_maxIterations = value;
 			}
-			
+
 			///run the path ROTplanner
 			///this will always populate path to be the path we need to travel
-			void run(
+			Planning::InterpolatedPath* run(
 					const Geometry2d::Point& start,
-					const float angle, 
-					const Geometry2d::Point& vel, 
+					const float angle,
+					const Geometry2d::Point& vel,
 					const MotionConstraints &motionConstraints,
-					const Geometry2d::CompositeShape* obstacles, 
-					Planning::Path &path);
-			
+					const Geometry2d::CompositeShape* obstacles);
+
 			/** returns the length of the best position planned path */
 			float fixedPathLength() const { return _bestPath.length(); }
-			
+
 	protected:
 		MotionConstraints _motionConstraints;
 
 		FixedStepTree _fixedStepTree0;
 		FixedStepTree _fixedStepTree1;
-		
+
 		/** best goal point */
 		Geometry2d::Point _bestGoal;
-		
+
 		///best planned path
 		///this is a fixed step path
-		Planning::Path _bestPath;
-		
+		Planning::InterpolatedPath _bestPath;
```

Comment:
I like that you've changed the Planner's run() method to return a new path rather than changing an existing one in place.  Is there much use in keeping a copy of the path in `_bestPath` here?